src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java

```diff
@@ -557,8 +557,9 @@ protected void removeSignaturesFromTransaction(BtcTransaction pegoutBtcTx, Feder
             Script inputRedeemScript = getRedeemScriptFromInput(txInput);
             logger.trace("[removeSignaturesFromTransaction] input {} scriptSig {}", inputIndex, pegoutBtcTx.getInput(inputIndex).getScriptSig());
             logger.trace("[removeSignaturesFromTransaction] input {} redeem script {}", inputIndex, inputRedeemScript);
-
-            txInput.setScriptSig(createBaseInputScriptThatSpendsFromTheFederation(spendingFed, inputRedeemScript));
+            if (!BitcoinUtils.inputHasWitness(pegoutBtcTx, inputIndex)) {
+                txInput.setScriptSig(createBaseInputScriptThatSpendsFromTheFederation(spendingFed, inputRedeemScript));
```

Comment:
Is it possible to have p2wsh and p2sh inputs at the same time? I guess it cannot. If so, you should validate if the pegout has a witness before calling this method and skip calling it.

Reply:
The witness depends on the input, not on the tx.

```
public static boolean inputHasWitness(BtcTransaction btcTx, int inputIndex) {
        TransactionWitness inputWitness = btcTx.getWitness(inputIndex);
        return !inputWitness.equals(TransactionWitness.getEmpty());
```